### PR TITLE
Don't run an analysis if we already have the points and don't need the results (#1880)

### PR DESCRIPTION
**Pull request by hbenl**

### CODE REVIEW

File: src/devtools/client/debugger/src/components/SecondaryPanes/Breakpoints/BreakpointTimeline.js [hunk 1]
```diff
@@ -87,32 +82,32 @@ function BreakpointTimeline({
     <div className="breakpoint-navigation-timeline-container">
       <div
         className={classnames("breakpoint-navigation-timeline", { dimmed: shouldDim })}
-        ref={timelineNode}
```
Comment (jasonLaster) [inline src/devtools/client/debugger/src/components/SecondaryPanes/Breakpoints/BreakpointTimeline.js:90]: this is great

File: src/devtools/client/debugger/src/components/SecondaryPanes/Breakpoints/BreakpointTimelinePoint.js [hunk 2]
```diff
@@ -25,45 +25,17 @@ function hasSecondaryHighlighted({ hoveredItem, breakpoint }) {
...
-  const startTime = zoomRegion.startTime;
-  const endTime = zoomRegion.endTime;
-
-  // Percent offset values are [0, 100].
-  const unadjustedLeftPercentOffset = ((point.time - startTime) / (endTime - startTime)) * 100;
-  const unadjustedLeftPixelOffset = (unadjustedLeftPercentOffset / 100) * timelineWidth;
-  const leftPixelOffset = unadjustedLeftPixelOffset - markerWidth / 2;
-  const leftPercentOffset = (leftPixelOffset * 100) / timelineWidth;
-  return leftPercentOffset;
-}
```
Comment (jasonLaster) [inline src/devtools/client/debugger/src/components/SecondaryPanes/Breakpoints/BreakpointTimelinePoint.js:40]: don't know how you're getting rid of this math, but i like it

File: src/protocol/logpoint.ts [hunk 3]
```diff
@@ -93,14 +98,20 @@ async function showPrimitiveLogpoints(
...
-    PointHandlers.onPoints(points, { locations });
+function saveLogpointHits(
+  points: PointDescription[],
+  results: AnalysisEntry[],
+  locations: Location[],
+  condition: string
+) {
+  if (condition) {
+    points = points.filter(point =>
+      results.some(result => result.key === point.point && result.value.time === point.time)
```
Comment (jasonLaster) [inline src/protocol/logpoint.ts:109]: what is this doing?

File: src/ui/actions/app.ts [hunk 4]
```diff
@@ -92,19 +90,6 @@ export function setupApp(recordingId: RecordingId, store: UIStore) {
...
-    locations.forEach(
-      (location: Location | null) => location && store.dispatch(setAnalysisPoints(points, location))
-    );
-  };
-
-  PointHandlers.addPendingNotification = (location: any) => {
-    store.dispatch(setPendingNotification(location));
-  };
-}
-
```
Comment (jasonLaster) [inline src/ui/actions/app.ts:107]: glad we're moving this

File: src/ui/reducers/app.ts [hunk 5]
```diff
@@ -85,7 +85,7 @@ export default function update(
     }
 
     case "set_analysis_points": {
```
Comment (jasonLaster) [inline src/ui/reducers/app.ts:87]: do you think it's time to add an analysis reducer?

**Review by jasonLaster — COMMENTED** [review]
Left some thoughts. Will test soon

File: src/devtools/client/debugger/src/components/SecondaryPanes/Breakpoints/BreakpointTimelinePoint.js (hunk 2 above)
Comment (hbenl) [reply src/devtools/client/debugger/src/components/SecondaryPanes/Breakpoints/BreakpointTimelinePoint.js:40]: [That's how](https://github.com/hbenl/webreplay-devtools/blob/55ffa9c433bd82eb2e31faa85ff5b6fe6c06c904/src/devtools/client/debugger/src/components/SecondaryPanes/Breakpoints/BreakpointTimelinePoint.js#L67). CSS is awesome.

File: src/protocol/logpoint.ts (hunk 3 above)
Comment (hbenl) [reply src/protocol/logpoint.ts:109]: Filtering out the `PointDescription`s referring to points where the condition is false.

File: src/ui/reducers/app.ts (hunk 5 above)
Comment (hbenl) [reply src/ui/reducers/app.ts:87]: Let's do that when it gets more complex.

**Comment by jasonLaster** [comment]
works for me

Sent via Superhuman ( https://sprh.mn/?vip=jason.laster.11@gmail.com )

File: src/devtools/client/debugger/src/components/SecondaryPanes/Breakpoints/BreakpointTimelinePoint.js [hunk 6]
```diff
@@ -89,7 +64,7 @@ function BreakpointTimelinePoint({
       })}
       title={`${index + 1}/${analysisPoints.length}`}
       onClick={() => seek(point.point, point.time, true)}
-      style={{ left: `${leftPercentOffset}%` }}
+      style={{ left: `calc(${leftPercentOffset}% - ${pointWidth / 2}px)` }}
```
Comment (jaril) [inline src/devtools/client/debugger/src/components/SecondaryPanes/Breakpoints/BreakpointTimelinePoint.js:67]: I think we could do this even simpler. To get the same half-of-width-displacement effect:
```
.breakpoint-navigation-timeline-point {
    transform: translateX(60px);
}
```
That way JS doesn't ever need to know about `pointWidth` in the first place. Reduces the code here to just `style={{ left: `${leftPercentOffset}%` }}`

**Review by jaril — COMMENTED** [review]
Left two comments on minor nits, but looks good!

File: src/devtools/client/debugger/src/components/SecondaryPanes/Breakpoints/BreakpointTimelinePoint.js (hunk 6 above)
Comment (hbenl) [reply src/devtools/client/debugger/src/components/SecondaryPanes/Breakpoints/BreakpointTimelinePoint.js:67]: That was my first idea, but then I saw that the javascript variable is used in 2 other places, so it's probably a good idea to have a single source of truth here.

**Pull request merged**